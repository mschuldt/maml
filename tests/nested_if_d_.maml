arduino = Arduino(desktop=True)

#each block will will be executed in the blockchain many times before 
#the next one is sent so each of them delays while the next block
#is sent to the vm

@block(once)
def test():
    if 1:
        if 2:
            print_i(1)
        if 0:
            print_i(2)
        else:
            if 2:
                if 0:
                    print_i(3)
                elif 2:
                    print_i(4)
            else:
                print_i(4)

            if 2:
                if 0:
                    print_i(3)
                elif 2:
                    print_i(4)
            else:
                print_i(4)
    else:
        print_i(5)

    die()

arduino.send(test)

