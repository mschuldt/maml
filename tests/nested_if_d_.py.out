Initializing avm...
Ready.

1
4
4
bye.
